pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -383,16 +385,18 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
               }
             }
 
+            int partitionSeq = 0;
```

Comment:
Actually, there's a bit better approach :) 

1. Check the `partitionToSegments.keys() - List<List<Integer>>` and validate if the length of the value list is all 1 (this means that the data is correctly partitioned).
2. If so, we can directly use the partition value. If not, we can fall back to the partitionSeq to avoid too long naming.

This approach will try to put the correct information at the best effort.

Reply:
Discussed offline again, we decided to change back to the initial solution.